ctx.go

```diff
@@ -25,17 +25,17 @@ type ctxKey struct{}
 //     l.UpdateContext(func(c Context) Context {
 //         return c.Str("bar", "baz")
 //     })
-func (l *Logger) WithContext(ctx context.Context) context.Context {
+func (l Logger) WithContext(ctx context.Context) context.Context {
 	if lp, ok := ctx.Value(ctxKey{}).(*Logger); ok {
-		if lp == l {
+		if lp == &l {
```

Comment:
So this change makes no sense. Value receiver methods get a new copy of `l` every time they are called. So here you are taking an address of this temporary copied value and comparing it with context pointer. This will always be false. This has been fixed by removing this check in #499.